src/gauging.jl

```diff
@@ -59,172 +86,96 @@ function vidal_gauge(
     ψvdst = replaceinds(ψvdst, edge_ind, edge_ind_sim)
     ψvdst = replaceinds(ψvdst * V, commoninds(V, S), new_edge_ind)
 
-    setindex_preserve_graph!(ψ_vidal, ψvsrc, vsrc)
-    setindex_preserve_graph!(ψ_vidal, ψvdst, vdst)
+    setindex_preserve_graph!(updated_ψ, ψvsrc, vsrc)
+    setindex_preserve_graph!(updated_ψ, ψvdst, vdst)
 
     S = replaceinds(
       S,
       [commoninds(S, U)..., commoninds(S, V)...] =>
         [new_edge_ind..., prime(new_edge_ind)...],
     )
-    bond_tensors[e] = S
+    updated_bond_tensors[e] = S
   end
 
-  return ψ_vidal, bond_tensors
+  return VidalITensorNetwork(updated_ψ, updated_bond_tensors)
 end
 
-"""Use an ITensorNetwork ψ in the symmetric gauge and its mts to put ψ into the vidal gauge. Return the bond tensors and ψ_vidal."""
-function vidal_gauge(
-  ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  edges=NamedGraphs.edges(ψ),
-  svd_kwargs...,
+function update(
+  ψv::VidalITensorNetwork;
+  (cache!)=nothing,
+  cache_update_kwargs=default_cache_update_kwargs(cache!),
+  kwargs...,
```

Comment:
I don't see why this version and the version above that accepts the cache as an argument both need to exist, I'm not really a fan of the function above anyway. Can these be combined?

I'm also losing track of the logic behind what the cache here is being used for, since I thought the bond tensors of the Vidal gauge play the role of the BP cache.